core/index/index_writer.hpp

```diff
@@ -314,7 +314,7 @@ class index_writer : private util::noncopyable {
     // the segment_context used for storing changes (lazy-initialized)
     active_segment_context segment_;
     // segment_.ctx().use_count() at constructor/destructor time must equal
-    uint64_t segment_use_count_{0};
+    long segment_use_count_{0};
```

Comment:
Why signed type? Do we expect negative values?

Reply:
No, but use_count return long, I don't want to make cast
https://stackoverflow.com/questions/36370180/why-does-shared-ptrtuse-count-return-a-long-instead-of-an-unsigned-type

Reply:
> Is i possible to make a test to reproduce the issue here?

Maybe... But I still want to remove this checks, so I don't think we need it